src/crud.py

```diff
@@ -0,0 +1,81 @@
+from typing import Optional
+from sqlalchemy.orm import Session
+from . import models, schemas
+
+
+def get_user(db: Session, address: str):
+    return db.query(models.User).filter(models.User.address == address).first()
+
+def create_user(db: Session, user: schemas.UserCreation):
+    db_user = models.User(**user.model_dump())
+    db.add(db_user)
+    db.commit()
+    db.refresh(db_user)
+    return db_user
+
+def get_contracts_by_user(db: Session, user_address: str):
+    user = get_user(db, user_address)
+    if not user:
+        return None
+    return user.contracts
+
+
+def get_contract(db: Session, address: str):
+    return db.query(models.Contract).filter(models.Contract.address == address).first()
+
+
+def get_entrypoints(db: Session, contract_address: str):
+    contract = get_contract(db, contract_address)
+    return contract.entrypoints if contract else []
+
+def get_entrypoint(db: Session, contract_address: str, name: str) -> Optional[models.Entrypoint]:
+    entrypoints = get_entrypoints(db, contract_address)
+    entrypoint = list(filter(lambda e: e.name == name, entrypoints))
+    if len(entrypoint) == 0:
+        return None
```

Comment:
Are you using mypy or another static analysis tool? I'm not, currently. We can decide to make it mandatory, but if we don't, I'd say that raising exceptions is better than returning None and risking to miss it.

Reply:
And we can raise exception 👍